lessons/unittest/original/module.py

```diff
@@ -0,0 +1,35 @@
+''' 
+  A class we can use to test unittesting
+'''
+
+import numpy as np
+
+
+class Var(object):
+  ''' A container class for numpy arrays and some meta data '''
+
+  def __init__(self, name, data, axes=None, units=None):
+    ''' construct variable object with meta data '''
+    self.name = name
+    self.units = units
+    self.axes = axes
+    self.data = data
+    
+  def __iadd__(self, data):
+    ''' overload in-place addition function: add number or arrays '''
+    self.data += data
+    return self
+  
+  def __add__(self, other):
+    ''' overload addition function: add arrays and create new variable '''
+    # check input
+    if not isinstance(other, Var): raise Exception
```

Comment:
Returning that value does produce an exception without you having to do anything special:
```python
In [16]: class A:
   ....:     def __init__(self, i):
   ....:         self.data = i
   ....:     def __add__(self, other):
   ....:         if isinstance(other, int):
   ....:             return A(self.data + other)
   ....:         else:
   ....:             return NotImplemented
   ....:         

In [17]: A(234)
Out[17]: <__main__.A at 0x7f3f4d0817b8>

In [18]: A(234)+2
Out[18]: <__main__.A at 0x7f3f4d081748>

In [19]: A(234)+'sdf'
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-0312a499262d> in <module>()
----> 1 A(234)+'sdf'

TypeError: unsupported operand type(s) for +: 'A' and 'str'
```